services/PredictionService.py

```diff
@@ -1,39 +1,23 @@
-from datetime import datetime
-from sqlalchemy.sql.sqltypes import String
 from controllers import db
-from models.AccountModel import Account
-from models.DataModel import Data
-from models.PredictionModel import Prediction
-from models.QAModel import QAModel
-from models.VisitorModel import Visitor
-from services.DataService import DataService
-from services.QAModelService import QAModelService
-import string
+from models import Prediction
 
 
 class PredictionService():
-    db_conn = db
-    data_service = DataService()
-    model_service = QAModelService()
 
     def create_prediction(self, prediction:Prediction)->Prediction:
-        self.db_conn.session.add(prediction)
-        self.db_conn.session.commit()
-        saved_prediction = Prediction.query.filter(Prediction.uuid == prediction.uuid).first()
+        db.session.add(prediction)
+        db.session.commit()
+        saved_prediction = Prediction.query.filter(Prediction.id == prediction.id).first()
         return saved_prediction
 
-    def update_prediction(self, prediction:Prediction)->Prediction:
-        prediction_retrieve = Prediction.query.filter(Prediction.uuid == prediction.uuid).first()
-        prediction_retrieve.prediction = prediction.prediction
+    def update_prediction(self, prediction: Prediction) -> Prediction:
+        prediction_retrieve = Prediction.query.filter(Prediction.id == prediction.id).first()
         prediction_retrieve.alt_answer = prediction.alt_answer
         prediction_retrieve.is_corrected = prediction.is_corrected
-        prediction_retrieve.visitor_id = prediction.visitor_id
-        prediction_retrieve.model_id = prediction.model_id
-        prediction_retrieve.datum_id = prediction.datum_id
```

Comment:
The only fields that are updatable are alt_answer and is_correct.  We only need to populate those two fields before saving.  That is something we should consider in all of our updates.  We should only expose the fields that we will allow to be updated.

Reply:
Yep, my mistake, looks great then!